dbt-snowflake/src/dbt/include/snowflake/macros/relations/dynamic_table/create.sql

```diff
@@ -1,85 +1,14 @@
 {% macro snowflake__get_create_dynamic_table_as_sql(relation, sql) -%}
-{#-
---  Produce DDL that creates a dynamic table
---
---  Args:
---  - relation: Union[SnowflakeRelation, str]
---      - SnowflakeRelation - required for relation.render()
---      - str - is already the rendered relation name
---  - sql: str - the code defining the model
---  Globals:
---  - config: NodeConfig - contains the attribution required to produce a SnowflakeDynamicTableConfig
---  Returns:
---      A valid DDL statement which will result in a new dynamic table.
--#}
 
+    {%- set catalog_relation = adapter.build_catalog_relation(config.model) -%}
     {%- set dynamic_table = relation.from_config(config.model) -%}
 
-    {%- if dynamic_table.catalog.table_format == 'iceberg' -%}
-        {{ _get_create_dynamic_iceberg_table_as_sql(dynamic_table, relation, sql) }}
+    {%- if catalog_relation.catalog_type == 'NATIVE' -%}
+        {{ snowflake__create_dynamic_table_standard_sql(dynamic_table, relation, compiled_code) }}
+    {%- elif catalog_relation.catalog_type == 'ICEBERG_MANAGED' -%}
+        {{ snowflake__create_dynamic_table_iceberg_managed_sql(dynamic_table, relation, compiled_code) }}
     {%- else -%}
-        {{ _get_create_dynamic_standard_table_as_sql(dynamic_table, relation, sql) }}
+        {% do exceptions.raise_compiler_error('Unexpected model config for: ' ~ relation) %}
     {%- endif -%}
 
 {%- endmacro %}
-
-
-{% macro _get_create_dynamic_standard_table_as_sql(dynamic_table, relation, sql) -%}
```

Comment:
nit: I think moving these to separate files may actually make it harder to reason about this code as you'll have to swap between three files to reason about diffs / make updates. Down to try it this way for now but worth a thought

Reply:
I can push them back together if desired. I was trying to follow the "one macro per file" thing that we vaguely discussed in the past. But we can move that up one layer to something like command (create table...) and we wouldn't really lose that much discoverability.